core/node/api_server/src/tx_sender/tests/send_tx.rs

```diff
@@ -130,10 +129,9 @@ async fn fee_validation_errors() {
     let l2_chain_id = L2ChainId::default();
     let tx_executor = SandboxExecutor::mock(MockOneshotExecutor::default()).await;
     let (tx_sender, _) = create_test_tx_sender(pool.clone(), l2_chain_id, tx_executor).await;
-    let fee_input = MockBatchFeeParamsProvider::default()
-        .get_batch_fee_input_scaled(1.0, 1.0)
-        .await
-        .unwrap();
+    let fee_params_provider: Arc<dyn BatchFeeModelInputProvider> =
+        Arc::new(MockBatchFeeParamsProvider::default());
```

Comment:
Nit: You can create an immediate reference here `let fee_params_provider: &dyn BatchFeeModelInputProvider = ..`; there's no need to wrap a provider in `Arc`.